### PR TITLE
cptofs: allow to override owner/group when copying files

### PR DESCRIPTION
**Pull request by Mic92**
cptofs is useful to write images without using virtual machines or having root access. This option allows to also write content as a different owner/group.

### CODE REVIEW

File: tools/lkl/cptofs.c [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#define _GNU_SOURCE
```
Comment (Mic92) [inline tools/lkl/cptofs.c:1]: Not sure if this is a problem? What operating systems do we need to support here?

File: tools/lkl/cptofs.c [hunk 2]
```diff
@@ -111,22 +121,29 @@ static int open_src(const char *path)
...
 	if (cptofs)
-		fd = lkl_sys_open(path, LKL_O_RDWR | LKL_O_TRUNC | LKL_O_CREAT,
-				  mode);
+		fd = lkl_sys_open(path, LKL_O_RDWR | LKL_O_TRUNC | LKL_O_CREAT, mode);
 	else
 		fd = open(path, O_RDWR | O_TRUNC | O_CREAT, mode);
 
+	if (owner != (uid_t)-1 && group != (gid_t)-1) {
+		if (cptofs)
+			ret = lkl_sys_fchown(fd, owner, group);
```
Comment (cbrewster) [inline tools/lkl/cptofs.c:136]: Should `ret` be checked here and below?

**Comment by tavip** [comment]
@Mic92 sorry for catching this late, there are a few coding style issues.

**Comment by Mic92** [comment]
Should be fine now:

```
$ scripts/checkpatch.pl tools/lkl/cptofs.c
WARNING: Missing or malformed SPDX-License-Identifier tag in line 1
#1: FILE: tools/lkl/cptofs.c:1:
+#ifdef __FreeBSD__

total: 0 errors, 1 warnings, 693 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

tools/lkl/cptofs.c has style problems, please review.

NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.
```

**Pull request merged**